src/root/usr/bin/run-postgresql

```diff
@@ -20,8 +20,14 @@ generate_postgresql_config
 # Is this brand new data volume?
 PG_INITIALIZED=false
 
-if [ ! -f "$PGDATA/postgresql.conf" ]; then
-  initialize_database
+if [ ! -f "${PGDATA}/postgresql.conf" ]; then
+  # Minimize the race condition of unfinished DB initialization vs pod restart
+  # First initialize the DB into temporary directory
+  TEMP_PGDATA=$(mktemp -d $(printf '%0.sX' {1..32}))
+  initialize_database "${TEMP_PGDATA}"
+  # If DB initialization succeeded, copy content of temporary directory to
+  # final PGDATA, remove the temporary directory & set PG_INITIALIZED flag
+  cp -prT "${TEMP_PGDATA}" "${PGDATA}" && rm -rf "${TEMP_PGDATA}"
```

Comment:
I really like the hack point of this hack, thank you for the idea!

I sort of doubt that **if** I/O is blocking us and causes unreasonable delays so container timeouts **/bin/cp** will make it that better (+ even doing `rm -rf` isn't for free).  Would it be possible to rather work with `mv` and also fix the `set_pgdata` function?

Reply:
@praiskup Will change _cp_ to _mv_, but what you mean under _also fix the ```set_pgdata``` function_? Just change the name, or should it also use some temporary dir?

Reply:
I was thinking about `mv "$TEMP_PGDATA" "$PGDATA"`, but for this we would have to have non-existing "$PGDATA" (so we'd have to work on `set_pgdata` a bit, it deserves it anyways).

But if there will be safe `mv` statement in subdirectory (I think) it will be good enough as well.

Reply:
Changed to use ```mv``` instead.